client/state/selectors/get-settings-url.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Internal dependencies
+ */
+import { getSiteSlug, getSiteAdminUrl } from 'calypso/state/sites/selectors';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import isSiteWPCOM from 'calypso/state/selectors/is-site-wpcom';
+
+/**
+ * Type dependencies
+ */
+import type { AppState } from 'calypso/types';
+
+type Sections = 'general';
+
+function getSettingsUrl(
+	state: AppState,
+	siteId: number | null,
+	section: Sections
+): string | null {
+	if ( ! siteId ) {
+		return null;
+	}
+	const siteSlug = getSiteSlug( state, siteId );
+	const siteAdminUrl = getSiteAdminUrl( state, siteId );
+	const siteIsWPCOM = isSiteWPCOM( state, siteId );
+	if ( isJetpackCloud() && ! siteIsWPCOM ) {
+		if ( section === 'general' ) {
+			return `${ siteAdminUrl }options-general.php`;
+		}
+	} else if ( siteIsWPCOM ) {
+		if ( section === 'general' ) {
+			return `/settings/general/${ siteSlug }`;
+		}
+	}
```

Comment:
I think you missed the Jetpack site in WordPress.com case.

Reply:
Oh, that'd make sense wouldn't it? 😅

Reply:
Resolved in b98d32a. Can you retest?